pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -74,23 +74,36 @@ const (
 
 const (
 	cephOSDStart = `
-function sigterm() {
-	echo "SIGTERM received"
+RESTART_INTERVAL=%d
+child_pid=""
+
+function handle_signal() {
+	echo "Signal received"
+	# kill the child process is its still running
+	kill -TERM "$child_pid"
 	exit
 }
 
-trap sigterm SIGTERM
+trap handle_signal SIGTERM
 
-%s %s & wait
+%s %s &
+child_pid="$!"
+wait "$child_pid"
```

Comment:
Where is the double wait? In the case of sigterm, this may exit the bash process before the child process is stopped, leaving it to zombie.